Dev/Pastebin/pastebin.3m.sh

```diff
@@ -0,0 +1,150 @@
+# <bitbar.image>https://i.ibb.co/cym797V/Pastebin2.png</bitbar.image>
+# <bitbar.abouturl>https://github.com/LivelyCarpet87/BitBar-Pastebin#bitbar-pastebin</bitbar.abouturl>
+#
+
+#Requires Configuration Here
+
+#Pastebin developer key
+dev_key=""
+
+#Pastebin user key
+usr_key=""
+
+#Save-To Directory
+#Saved pastes will be saved to this directory
+saveDir=""
+
+#Enable Deletion of pastes
+# 0=disabled | 1=enabled
+deleteEnabled=1
+
+#End Configuration
+
+#Constants
+#Pastebin API List Paste URL
+list_paste_url="https://pastebin.com/api/api_post.php"
+#Pastebin API Get Paste URL
+get_paste_url="https://pastebin.com/api/api_raw.php"
+
+
+#empty pastes arrays
+pasteNameArr=()
+pasteKeyArr=()
+
+#Initialize the temp directory if it does not exist
+if [ ! -d /tmp/pastebinReader ]
+then
+mkdir /tmp/pastebinReader
+fi
+
+#clear the temp directory
+rm -f /tmp/pastebinReader/*
+
+
+#get a list of all the pastes
+
+queryResults=$(curl --silent --connect-timeout 15 --speed-time 15 --speed-limit 500  -X POST -d "api_option=list&api_user_key=$usr_key&api_dev_key=$dev_key" $list_paste_url)
+
+#parse the paste titles and keys
+
+titles=$(echo "$queryResults" | grep -E "<paste_title>([^\r]*)" --context=0 | sed s+'<paste_title></paste_title>'+Untitled+g | sed s+\<paste_title\>++g | sed s+'</paste_title>'++g |tr '\r' ' ')
+
+pasteURLs=$(echo "$queryResults" | grep -E "<paste_key>(\S*)</paste_key>" --context=0 | sed s+\<paste_key\>++g | sed s+'</paste_key>'++g |tr '\r' ' ')
+
+SAVEIFS=$IFS
+IFS=$'\n\r'
+#shellcheck disable=SC2206
+pasteNameArr=($titles)
+#shellcheck disable=SC2206
+pasteKeyArr=($pasteURLs)
+IFS=$SAVEIFS
+
+totalPastes=$((${#pasteNameArr[@]}))
+
+
+
+# test if the request response was valid
+if [[ $(echo "$queryResults" | grep "paste" -c) -gt 0 || $(echo "$queryResults" | grep "No pastes found" -c) == 1 ]]
+then
+
+echo "Found: $totalPastes"
+
+i=0
+while [[ $i < $totalPastes ]]
+do
+printf '%s\n' "$((i + 1)): ${pasteNameArr[$i]} | length=7 dropdown=false"
+i=$((i + 1))
+done
+
+echo "---"
+i=0
+while [[ $i < $totalPastes ]]
+do
+#print paste title
+printf '%s\n' "$((i + 1)): ${pasteNameArr[$i]} |href=https://pastebin.com/${pasteKeyArr[$i]}"
+
+#get and display paste content with slight modifications to prevent breaking Bitbar display
+pasteContent="$(curl --silent -X POST --connect-timeout 15 --speed-time 15 --speed-limit 500 -d "api_option=show_paste&api_user_key=$usr_key&api_dev_key=$dev_key&api_paste_key=${pasteKeyArr[$i]}"  $get_paste_url)"
+
+echo "-- $(echo "$pasteContent" | tr '\n' '$' |  tr '\r' '$' | sed 's/\$\$/; /g' | sed 's/;;/; /g')"
+
+#save paste contents temporarily until next refresh
+
+fname=$(echo "${pasteNameArr[$i]}_${pasteKeyArr[$i]}"| tr ' ' '_'| tr '' '_'| tr '#' 'H'|tr '$' 'S'| tr '&' '+'| tr -d '>'| tr -d '<'| tr -d '[' | tr -d ']'| tr -d '(' | tr -d ')' | tr -d '`' | tr -d '"' | tr -d '\''| tr -d '\' | tr -d '|' | tr -d ';' |tr -d '*' | tr -d '?'| tr -d '~' | tr -d '/')
```

Comment:
```
name=$(echo "${pasteNameArr[$i]}_${pasteKeyArr[$i]}"| tr ' ' '_'| tr '' '_'| tr '#' 'H'|tr '$' 'S'| tr '&' '+'| tr -d '>'| tr -d '<'| tr -d '[' | tr -d ']'| tr -d '(' | tr -d ')' | tr -d '`' | tr -d '"' | tr -d '\''| tr -d '\' | tr -d '|' | tr -d ';' |tr -d '*' | tr -d '?'| tr -d '~' | tr -d '/')
                                                                                                                                                                                                                     ^-- SC1003: Want to escape a single quote? echo 'This is how it'\''s done'.
```

in this case, just do `tr -d \'`

Reply:
Thank you. That one stumped me for quite a while.